tests/Fixer/PhpUnit/PhpUnitDataProviderStaticFixerTest.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\PhpUnit;
+
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @internal
+ *
+ * @property ConfigurableFixerInterface $fixer
+ *
+ * @covers \PhpCsFixer\Fixer\PhpUnit\PhpUnitDataProviderStaticFixer
+ */
+final class PhpUnitDataProviderStaticFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider provideFixCases
+     */
+    public function testFix(string $expected, ?string $input = null): void
+    {
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return iterable<array{0: string, 1?: null|string, 2?: array<string, bool>}>
+     */
+    public static function provideFixCases(): iterable
+    {
+        yield 'do not fix when containing dynamic calls by default' => [
+            '<?php
+class FooTest extends TestCase {
+    /**
+     * @dataProvider provideFoo1Cases
+     */
+    public function testFoo1() {}
+    public function provideFoo1Cases() { $this->init(); }
+}',
+        ];
+
+        yield 'fix single' => [
+            '<?php
+class FooTest extends TestCase {
+    /**
+     * @dataProvider provideFooCases
+     */
+    public function testFoo() {}
+    public static function provideFooCases() { $x->getData(); }
+}',
+            '<?php
+class FooTest extends TestCase {
+    /**
+     * @dataProvider provideFooCases
+     */
+    public function testFoo() {}
+    public function provideFooCases() { $x->getData(); }
+}',
+        ];
+
+        yield 'fix multiple' => [
+            '<?php
+class FooTest extends TestCase {
+    /** @dataProvider provider1 */
+    public function testFoo1() {}
+    /** @dataProvider provider2 */
+    public function testFoo2() {}
+    /** @dataProvider provider3 */
+    public function testFoo13() {}
+    public static function provider1() {}
```

Comment:
it's testing `fix multiple`, but actually it adds **single** `static` keyword.
let's add more, to make sure adding multiple slices works properly: https://github.com/werlos/PHP-CS-Fixer/pull/8